File: C#/DesignPatterns/Behavioral/Observer/DesignPatterns.BryanHansen.Observer/TabletClient.cs
using System;

namespace DesignPatterns.BryanHansen.Observer
{
    public class TabletClient : Observer
    {
        private readonly Subject _subject;

        public TabletClient(Subject subject)
        {
            _subject = subject;
            _subject.Attach(this);
        }

        public void AddMessage(string message)
        {
            _subject.SetState($"{message} - sent from tablet");
        }

        public override void Update()
        {
            Console.WriteLine($"Tablet stream: {_subject.GetState()}");
        }
    }
}

File: Java/src/main/java/behavioral/observer/messaging/TabletClient.java
package behavioral.observer.messaging;

public class TabletClient extends Observer {
    private final Subject subject;

    public TabletClient(Subject subject) {
        this.subject = subject;
        subject.attach(this);
    }

    public void addMessage(String message) {
        subject.setState(message + " - sent from tablet");
    }

    @Override
    void update() {
        System.out.println("Tablet stream: " + subject.getState());
    }
}
